_tour/variances.md

```diff
@@ -12,55 +12,99 @@ redirect_from: "/tutorials/tour/variances.html"
 
 Variance lets you control how type parameters behave with regards to subtyping. Scala supports variance annotations of type parameters of [generic classes](generic-classes.html), to allow them to be covariant, contravariant, or invariant if no annotations are used. The use of variance in the type system allows us to make intuitive connections between complex types.
 
+{% tabs variances_1 %}
+{% tab 'Scala 2 and 3' for=variances_1 %}
 ```scala mdoc
 class Foo[+A] // A covariant class
 class Bar[-A] // A contravariant class
 class Baz[A]  // An invariant class
 ```
+{% endtab %}
+{% endtabs %}
 
 ### Invariance
 
 By default, type parameters in Scala are invariant: subtyping relationships between the type parameters aren't reflected in the parameterized type. To explore why this works the way it does, we look at a simple parameterized type, the mutable box.
 
+{% tabs invariance_1 %}
+{% tab 'Scala 2 and 3' for=invariance_1 %}
 ```scala mdoc
 class Box[A](var content: A)
 ```
+{% endtab %}
+{% endtabs %}
 
 We're going to be putting values of type `Animal` in it. This type is defined as follows:
 
+{% tabs invariance_2 class=tabs-scala-version %}
+{% tab 'Scala 2' for=invariance_2 %}
 ```scala mdoc
 abstract class Animal {
   def name: String
 }
 case class Cat(name: String) extends Animal
 case class Dog(name: String) extends Animal
 ```
+{% endtab %}
+{% tab 'Scala 3' for=invariance_2 %}
+```scala
+abstract class Animal:
+  def name: String
+
+case class Cat(name: String) extends Animal
+case class Dog(name: String) extends Animal
+```
+{% endtab %}
+{% endtabs %}
 
 We can say that `Cat` is a subtype of `Animal`, and that `Dog` is also a subtype of `Animal`. That means that the following is well-typed:
 
+{% tabs invariance_3 %}
+{% tab 'Scala 2 and 3' for=invariance_3 %}
 ```scala mdoc
-  val myAnimal: Animal = Cat("Felix")
+val myAnimal: Animal = Cat("Felix")
 ```
+{% endtab %}
+{% endtabs %}
 
 What about boxes? Is `Box[Cat]` a subtype of `Box[Animal]`, like `Cat` is a subtype of `Animal`? At first sight, it looks like that may be plausible, but if we try to do that, the compiler will tell us we have an error:
 
+{% tabs invariance_4 class=tabs-scala-version %}
+{% tab 'Scala 2' for=invariance_4 %}
+```scala
```

Comment:
this will let you check with mdoc, but without name conflicts (it wraps code in `locally`)
```suggestion
```scala mdoc:nest
```